projects/ngrx.io/content/marketing/docs.md

```diff
@@ -2,14 +2,57 @@
 
-## Core Principles
-
-- State is a single, immutable data structure.
-- Components delegate responsibilities to side effects, which are handled in isolation.
-- Type-safety is promoted throughout the architecture with reliance on TypeScript's compiler for program correctness.
-- Actions and state are serializable to ensure state is predictably stored, rehydrated, and replayed.
-- Promotes the use of functional programming when building reactive applications.
-- Provide straightforward testing strategies for validation of functionality.
+## Why NgRx?
+
+NgRx is built for creating maintainable explicit applications, by storing single state and the use of actions in order to express state changes.
+
+
+### Serializability
+
+By normalizing state changes and pass them through observables, NgRx provides serializability and ensures state is predictably stored. This enables to save the state to an external storage, for example, `localStorage`.
+
+In addition, it also allows to inspect, download, upload, and dispatch actions, all from the [Store Devtools](guide/store-devtools).
+
+
+### Type Safety
+Type safety is promoted throughout the architecture with reliance on the TypeScript compiler for program correctness.
+
+
+### Encapsulation
+
+Using NgRx [Effects](guide/effects) and [Store](guide/store), any interaction with external resources side effects, like network requests, web socket and any business logic can be isolated from the UI. This isolation allows for more pure and simple components, and keep the single responsibility principle.
+
+### Testable
+
+Since NgRx is mostly based on pure functions like `selectors` and `reducers`, plus the ability to isolate side effects from the UI, testing becomes very straightforward.
+NgRx also provides tests setup like `provideMockStore` and `provideMockActions` for isolated tests, and a better test experience.
+
+### Performance
+
+NgRx is built on a single immutable data state, making change detection turn into a very easy task, which means [`OnPush`](https://angular.io/api/core/ChangeDetectionStrategy#OnPush) strategy becomes available.
+NgRx is also powered by memoized selector functions which optimize state query computations.
+
+## When Should I use NgRx
+
+In particular, you might use NgRx when you build an application with a lot of user interactions and multiple data sources, when managing state in services are no longer sufficient.
+
+A good substance that might answer the question "Do I need NgRx", is the 
+<a href="https://youtu.be/omnwu_etHTY" target="_blank">**SHARI**</a> principle:
+
+* **S**hared state is accessed by many components and services
+
+* **H**ydrated  state that is persisted and and hydrated from storage
```

Comment:
```suggestion
* **H**ydrated: state that is persisted and and rehydrated from external storage.
```

Reply:
IMHO, explain 'Hydrated' by using 're-hydrated' may cause endless recursion in some twisted dev brains.
Persisted and re-created (re-filled) might be a bit more human-friendly.

Also, there is a double 'and' typo.